### PR TITLE
KaDalu Operator

### PR DESCRIPTION
**Pull request by aravindavk**
- KaDalu Operator deploys required CSI drivers on start
- Operator listens to PV hosting Volume configurations via `kind:
  KadaluVolume`
- Added support for Replica1 and Replica3 hosting Volumes
- CSI provisioner can intelligently select from multiple hosting
  Volumes available

Signed-off-by: Aravinda VK <mail@aravindavk.in>

### CODE REVIEW

File: manifests/Replica3.client.vol.j2 [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+end-volume
+
+volume {{ volname }}-distribute
+    type cluster/distribute
+    option tier-hot-compact-frequency 604800
+    option tier-compact off
+    option watermark-low 75
+    option read-freq-threshold 0
+    option min-free-inodes 5%
+    option tier-query-limit 100
```
Comment (amarts) [inline manifests/Replica3.client.vol.j2]: tier is removed from glusterfs-6. I would say, no need of distribute at all in this graph.

File: manifests/Replica3.client.vol.j2 [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    option aggregate-size 128KB
+    option resync-failed-syncs-after-fsync off
+    option strict-write-ordering off
+    option strict-O_DIRECT off
+    option trickling-writes on
+    option cache-size 1MB
+    subvolumes {{ volname }}-md-cache
+end-volume
+
+volume {{ volname }}-io-cache
```
Comment (amarts) [inline manifests/Replica3.client.vol.j2]: lets remove this volume definition.

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,6 @@
 glustercs_csi.egg-info
 build
 dist
-manifests/*.yaml
\ No newline at end of file
+manifests/*.yaml
```
[the remark this answers is not in the document]
Comment (aravindavk) [reply .gitignore]: makes sense. I will rearrange the files/dirs

File: manifests/Replica3.client.vol.j2 (hunk 1 above)
Comment (aravindavk) [reply manifests/Replica3.client.vol.j2]: This volfile template is taken from Volume created using Gd2. I will work on the suggestion

**Pull request merged**